src/rslice.rs

```diff
@@ -173,6 +176,20 @@ impl<'a> rustls_str<'a> {
     }
 }
 
+impl<'a> fmt::Debug for rustls_str<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        unsafe {
+            let ptr = self.data as *const u8;
+            let raw = slice::from_raw_parts(ptr, self.len as usize + 1);
+            let cstr = CStr::from_bytes_with_nul_unchecked(raw);
```

Comment:
This was the best I could come up with but I'm sure that there's a better way

Reply:
This is not correct and Miri flags it. All of the `CStr` API's expect a null byte at the end, we don't have one, which I try to fake by making the array one larger than it actually is. 

I'll take a look at doing this another way.

Reply:
ah, I found a better way, I think